### ISSUE TITLE
[css-fonts] should supported tech() be validated during parsing or loading?

### ISSUE THREAD

**Issue by vitorroriz** [w3c/csswg-drafts#8793]
The way I interpreted `@font-face src tech()` from WPT tests, we should parse all valid keywords for `tech()` and then, at loading, check if the parsed tech is supported or not. That’s what WPT expects, for example, [here](https://github.com/web-platform-tests/wpt/blob/5e9760d203/css/css-fonts/parsing/font-face-src-format.html), otherwise a browser not supporting `color-colrv1`, for example, would fail some subtests.

However, [spec](https://www.w3.org/TR/css-fonts-4/#font-face-src-parsing) says:

> If a component value is parsed correctly and is of a font format or font tech that the UA supports, add it to the list of supported sources. If parsing a component value results in a parsing error or its format or tech are unsupported, do not add it to the list of supported sources.
> If there are no supported entries at the end of this process, the value for the src descriptor is a parse error.

For _unsupported_ tech to be a parsing error, we would need to validate for supported tech already during parsing, but then a browser not supporting, for example, color-colrv1 would fail the mentioned WPT subtest.

**Comment by cdoublev** [comment]
I think I am asking a similar question in #8795. A browser not supporting a property will fail a WPT subtest that uses this property in a container or feature query, if `<declaration>` does not match any syntactically valid declaration.

**Comment by jfkthame** [comment]
I agree it conflicts with the text about parsing. Tagged this for the CSSWG agenda in the hope we can get it clarified.

**Comment by svgeesus** [comment]
Does it though? It seems like a queue of operations:

1. does it parse
2. is it supported
3. can you fetch it
4. use it

So it is easily possible to have something that does parse and is not supported. Which means you don't try to fetch it.

I must be missing something here, this seems straightforward.

**Comment by jfkthame** [comment]
> It seems like a queue of operations:

>   1. does it parse
>   2. is it supported
>   3. can you fetch it
>   4. use it

That seems straightforward enough, yes.

But the issue is that the current spec text:

> If parsing a component value results in a parsing error **or its format or tech are unsupported**, do not add it to the list of supported sources.

> If there are no **supported** entries at the end of this process, the value for the [src](https://www.w3.org/TR/css-fonts-4/#descdef-font-face-src) descriptor is a **parse error**.

says (as I read it) that if the entries in the src descriptor are **parsed** successfully, but not **supported** by the UA, this results in a **parse error**. It shouldn't; the descriptor was *parsed* fine, it just resulted in a value that is not *supported* and therefore should not be *fetched*.

**Comment by svgeesus** [comment]
although we could usefully linkify formats and technologies
